boot/boot_images.cmake

```diff
@@ -46,12 +55,17 @@ set(_efisys_file  ${CMAKE_CURRENT_BINARY_DIR}/efisys.bin) # get_target_property(
 # - since its contents are included by mkisofs in the root of the ISO image,
 #   using the empty directory ensures that no extra unwanted files are added.
 #
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/bootfiles.sort "\
+set(ISO_SORT_FILE_DATA "\
 ${CMAKE_CURRENT_BINARY_DIR}/empty/boot.catalog 4
 ${_isoboot_file} 3
 ${_isobtrt_file} 2
-${_efisys_file} 1
 ")
+if(DEFINED EFI_PLATFORM_ID)
+  string(APPEND ISO_SORT_FILE_DATA "${_efisys_file} 1\n")
+endif()
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/bootfiles.sort "${ISO_SORT_FILE_DATA}")
+
+separate_arguments(ISO_BOOT_PARAMETERS WINDOWS_COMMAND "${ISO_BOOT_PARAMETERS}")
```

Comment:
```suggestion
separate_arguments(ISO_EFI_BOOT_PARAMS WINDOWS_COMMAND "${ISO_EFI_BOOT_PARAMS}")
```

Reply:
Question 2: Why do we need the `separate_arguments` ? (only place in all the ReactOS code). Is it because the `ISO_EFI_BOOT_PARAMS` were originally built (see above) "within a string"?